packages/authentication_repository/lib/src/models/user.dart

```diff
@@ -0,0 +1,35 @@
+import 'package:equatable/equatable.dart';
+
+/// {@template user}
+/// User model
+///
+/// [User.empty] represents an unauthenticated user.
+/// {@endtemplate}
+class User extends Equatable {
```

Comment:
I wouldn't name it the same as firebase auth user because it will make a lot of problems with conflicts later on. Usually, I name this `UserProfile` or `CurrentUser` note that firebase user returns empty user when a person is not logged in. Here, it is worth to think if you want to allow not registered user to use your app.

Reply:
I would also add here named constructoe such as User.fromFirebaseUser();

Reply:
All these comments are leading you to separate the model that you are getting from firebase from the app model. It might not be useful here but if the user had to have more fields in profile than firebase provides then you would have to have additional fields.

Reply:
Should I create seperate package called app_models and define all my app models there?